File: Coding Challenges/Leetcode/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l, r = 0, len(nums)

        while l<r:
            m = (l+r)//2
            if nums[m] == target: 
                l, r=m, m
                while l-1>=0 and nums[l-1] == target:
                    l-=1
                while r+1<len(nums) and nums[r+1] == target:
                    r+=1
                return [l, r]
            elif target > nums[m]:
                l = m+1
            else:
                r = m


        return [-1, -1]
      
# Runtime: 84 ms
# Memory Usage: 15.4 MB

File: Coding Challenges/Leetcode/34. Find First and Last Position of Element in Sorted Array.js
// Medium
var searchRange = function(nums, target) {
    let a=-1, b=-1, turned=false, len=nums.length, i=0;
    while(i<len){
      if(nums[i]===target && !turned){a=i, turned=true}
      else if(nums[i]===target){b=i}
      i++
    }
    if(a>-1 && b===-1) b=a
    return [a, b]
};

// Runtime: 80 ms, faster than 65.87% of JavaScript online submissions for Find First and Last Position of Element in Sorted Array.
